svc.go

```diff
@@ -88,6 +106,17 @@ func (as *AutoNATService) handleStream(s network.Stream) {
 	res.Type = pb.Message_DIAL_RESPONSE.Enum()
 	res.DialResponse = dr
 
+	// add an identity certificate
+	dr.DialerIdentityCertificate = new(pb.Message_DialerIdentityCertificate)
+	dr.DialerIdentityCertificate.PublicKey = as.cpk
+	sgn, err := as.sk.Sign(varint.ToUvarint(*req.Dial.Nonce))
```

Comment:
can we sign directly from `as.dialer.Peerstore().PrivKey(as.dialer.ID()).Sign(...)` rather than making a copy of the private key on the autonat object?
Keeping key material in one place seems preferable.